### PR TITLE
Preserve decoded arrays from filtering

### PR DESCRIPTION
**Pull request by thinkharderdev**
For kicks I took a crack at what preserving the decoded arrays might look like and this is what I came up with

### CODE REVIEW

File: parquet/src/arrow/arrow_reader/mod.rs [hunk 1]
```diff
@@ -353,21 +354,31 @@ impl ParquetRecordBatchReader {
...
+        batches.push(filter_record_batch(&batch, &filter)?);
+        filters.push(filter);
     }
 
-    let raw = RowSelection::from_filters(&filters);
-    Ok(match selection {
-        Some(selection) => selection.and(&raw),
-        None => raw,
-    })
+    let batch = RecordBatch::concat(&schema, &batches)?;
```
Comment (tustvold) [inline parquet/src/arrow/arrow_reader/mod.rs:374]: If you have a not very selective predicate, won't this massively balloon your memory consumption? As you will effectively hydrate the entire row group into a RecordBatch?

Comment (tustvold) [reply parquet/src/arrow/arrow_reader/mod.rs:374]: Perhaps we could have a hybrid approach where if the predicate isn't proving very selective, we bail out? :thinking:

Comment (thinkharderdev) [reply parquet/src/arrow/arrow_reader/mod.rs:374]: Not sure I follow. What would get pushed into the parquet reader?

Comment (tustvold) [reply parquet/src/arrow/arrow_reader/mod.rs:374]: If we've buffered too much in memory, which is highly likely with any non-selective predicate, the parquet reader could drop the buffer, and fall back to decoding it again

Comment (thinkharderdev) [reply parquet/src/arrow/arrow_reader/mod.rs:374]: > If we've buffered too much in memory, which is highly likely with any non-selective predicate, the parquet reader could drop the buffer, and fall back to decoding it again

Ah, yeah that makes sense

**Pull request closed without merging**